components/popover_menu_item/index.tsx

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ReactNode} from 'react';
+
+import {MenuItem, MenuItemProps, styled} from 'components/menu';
+
+interface Props extends MenuItemProps {
+    primaryLabel: ReactNode;
+    leadingElement?: ReactNode;
+    secondaryLabel?: ReactNode;
+    trailingElement?: ReactNode;
+    subMenuDetails?: ReactNode;
+}
+
+export function PopoverMenuItem(props: Props) {
+    const {primaryLabel, leadingElement, secondaryLabel, trailingElement, subMenuDetails, isDestructive, ...restProps} = props;
+
+    return (
+        <MenuItemStyled
+            disableRipple={true}
+            disableTouchRipple={true}
+            isDestructive={isDestructive}
+            {...restProps}
+        >
+            {leadingElement}
+            <div className='label-elements'>
+                {primaryLabel}
```

Comment:
Nitpicking a bit, but are these various prop names coming from the Compass design stuff? It seems a bit weird to me that we'd have separate props for `primaryLabel`/`secondaryLabel` and `subMenuDetails`/`trailingElement` when we could just have `label` and `trailingElement` props that you could pass a fragment into
```js
<PopoverMenuItem
    leadingElement={<LeadingElement/>}
    primaryLabel={<PrimaryLabel/>}
    secondaryLabel={<SecondaryLabel/>}
    subMenuDetails={<SubMenuDetails/>}
    trailingElement={<TrailingElement/>}
/>
// vs
<PopoverMenuItem
    leadingElement={<LeadingElement/>}
    label={
        <>
            <PrimaryLabel/>
            <SecondaryLabel/>
         </>
    }
    trailingElement={
        <>
            <SubMenuDetails/>
            <TrailingElement/>
        </>
    }
/>
```
Although that does end up being more lines vertically because of the fragments

Reply:
yes simplified the props as per above suggestion.